common/static/common/js/components/views/paging_footer.js

```diff
@@ -1,65 +1,69 @@
 
-        var PagingFooter = Backbone.View.extend({
-            events : {
-                "click .next-page-link": "nextPage",
-                "click .previous-page-link": "previousPage",
-                "change .page-number-input": "changePage"
-            },
+            var PagingFooter = Backbone.View.extend({
+                events : {
+                    "click .next-page-link": "nextPage",
+                    "click .previous-page-link": "previousPage",
+                    "change .page-number-input": "changePage"
+                },
 
-            initialize: function(options) {
-                var view = options.view,
-                    collection = view.collection;
-                this.view = view;
-                collection.bind('add', _.bind(this.render, this));
-                collection.bind('remove', _.bind(this.render, this));
-                collection.bind('reset', _.bind(this.render, this));
-                this.render();
-            },
+                initialize: function(options) {
+                    this.collection = options.collection;
+                    this.hideWhenOnePage = options.hideWhenOnePage || false;
+                    this.collection.bind('add', _.bind(this.render, this));
+                    this.collection.bind('remove', _.bind(this.render, this));
+                    this.collection.bind('reset', _.bind(this.render, this));
+                    this.render();
+                },
 
-            render: function() {
-                var view = this.view,
-                    collection = view.collection,
-                    currentPage = collection.currentPage,
-                    lastPage = collection.totalPages - 1;
-                this.$el.html(_.template(paging_footer_template, {
-                    current_page: collection.currentPage,
-                    total_pages: collection.totalPages
-                }));
-                this.$(".previous-page-link").toggleClass("is-disabled", currentPage === 0).attr('aria-disabled', currentPage === 0);;
-                this.$(".next-page-link").toggleClass("is-disabled", currentPage === lastPage).attr('aria-disabled', currentPage === lastPage);
-                return this;
-            },
+                render: function() {
+                    if (this.hideWhenOnePage) {
+                        if (this.collection.totalPages <= 1) {
+                            this.$el.addClass('hidden');
+                        } else if (this.$el.hasClass('hidden')) {
+                            this.$el.removeClass('hidden');
+                        }
+                    }
+                    this.$el.html(_.template(paging_footer_template, {
+                        current_page: this.collection.currentOneIndexPage(),
+                        total_pages: this.collection.totalPages
+                    }));
+                    var onFirstPage = !this.collection.hasPreviousPage();
```

Comment:
We typically declare all of our variables at the beginning of the function:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var

I forget with JSHint will catch this, but I have my PyCharm settings to report about this.